app_dart/lib/src/service/firestore.dart

```diff
@@ -27,4 +32,67 @@ class FirestoreService {
     );
     return FirestoreApi(client).projects.databases.documents;
   }
+
+  /// Writes [writes] to Firestore within a transaction.
+  ///
+  /// This is an atomic operation: either all writes succeed or all writes fail.
+  Future<CommitResponse> writeViaTransaction(List<Write> writes) async {
+    final ProjectsDatabasesDocumentsResource databasesDocumentsResource = await documentResource();
+    final BeginTransactionRequest beginTransactionRequest =
```

Comment:
Some databases have multiple commit modes. Is there anything in Firestore where you don't have to manage the commits manually like with with transactions?

Reply:
For a transaction operation, a commit is explicitly needed. There are other operations like `batchWrites`, which doesn't need a commit.